test/opengl/lut.cpp

```diff
@@ -46,30 +51,14 @@ int test_lut1d() {
     f0.realize(out0, target);
     out0.copy_to_host();
 
-    for (int c = 0; c != out0.extent(2); ++c) {
-        for (int y = 0; y != out0.extent(1); ++y) {
-            for (int x = 0; x != out0.extent(0); ++x) {
-                float expected = std::numeric_limits<float>::infinity();
-                switch (c) {
-                    case 0:
-                        expected = (float)(1 + x);
-                        break;
-                    case 1:
-                        expected = (float)(8 - x);
-                        break;
-                    case 2:
-                        expected = x > 3 ? 8.0f : 1.0f;
-                        break;
-                }
-                float result = out0(x, y, c);
-
-                if (result != expected) {
-                    fprintf(stderr, "Error at %d,%d,%d %f != %f\n", x, y, c, result, expected);
-                    return 1;
-                }
-            }
-        }
-    }
+    if (!Testing::check_result<float>(out0, [](int x, int y, int c) {
+            switch (c) {
+                case 0: return  (float)(1 + x);
+                case 1: return (float)(8 - x);
+                case 2: return (x > 3) ? 8.0f : 1.0f;
+                default: return std::numeric_limits<float>::infinity();
+            } }))
+        return 1;
```

Comment:
Nit: IMHO this would be clearer with explicit braces around the if clause. (I realize we don't come close to a uniform style on this in Halide, but generally, prefer explicit braces for all if/else/for blocks, even if single-statement.)